columnflow/tasks/framework/base.py

```diff
@@ -266,6 +266,197 @@ def has_obj(name):
+
+        # return either a tuple or only the first param, based on the *multiple* parameter
+        if multiple:
+            return law.util.make_tuple(param, cast=True)
+        elif param and isinstance(param, (list, tuple)):
+            return param[0]
+
+        return param
+
+    @classmethod
+    def resolve_config_default_and_groups(
+        cls,
+        task_params: dict[str, Any],
+        param: str | tuple[str] | None,
+        container: str | od.AuxDataMixin = "config_inst",
+        default_str: str | None = None,
+        groups_str: str | None = None,
+    ) -> tuple[str]:
+        """
+        Resolves tuple of inputs *param* by using groups identified via the *groups_str* from
+        an :py:class:`order.AuxDataMixin` *container* that maps a string to a tuple of strings.
+        If *param* is empty, the default identified by *default_str* from the *container* is returned.
+        If *default_str* points to a function, it should take *container* and *task_params* as an input.
+
+        Example:
+
+        .. code-block:: python
+            def resolve_param_values(params):
+                params["producers"] = ConfigTask.resolve_config_default_and_groups(
+                    params,
+                    params.get("producers"),
+                    container=params["config_inst"]
+                    default_str="default_producer",
+                    groups_str="producer_groups",
+                )
+
+            config_inst = od.Config(
+                id=0,
+                name="my_config",
+                aux={
+                    "default_producer": ["features_1", "my_producer_group"],
+                    "producer_groups": {"my_producer_group": ["features_2", "features_3"]},
+                },
+            )
+
+            params = {
+                "config_inst": config_inst,
+                "producer": None,
+            }
+            resolve_param_values(params)  # sets params["producers"] to ("features_1", "features_2", "features_3")
+        """
+        # get the container inst
+        if isinstance(container, str):
+            container = task_params.get(container, None)
+
+        if not container:
+            return param
+
+        # expand default when param is empty
+        if param in (None, law.NO_STR, tuple()):
+            param = container.x(default_str, None) if default_str else None
+
+            # allow default to be a function, taking task parameters as input
+            if isinstance(param, Callable):
+                param = param(container, **task_params)
+
+            if not param:
+                return ()
+
+        # expand to groups
+        if default_str and (param_groups := container.x(groups_str, {})):
+            values = []
+            for val in law.util.make_list(param):
+                values.extend(param_groups.get(val, [val]))
+            return tuple(values)
+
+        # return the param as is
+        return law.util.make_tuple(param)
```

Comment:
Mh, the group expansion only works for one level of nesting.

```python
if not param:  # moved one indentation level to the left
    return ()

# expand groups recursively
if default_str and (param_groups := container.x(groups_str, {})):
    values = []
    lookup = law.util.make_list(param)
    handled_groups = set()
    while lookup:
        value = lookup.pop(0)
        if value in param_groups:
            if value in handled_groups:
                raise Exception(f"definition of '{group_str}' contains self-references")
            lookup = law.util.make_list(param_groups[value]) + lookup
            handled_groups.add(value)
        else:
            values.append(value)
    param = values

return law.util.make_tuple(param)
````

Reply:
Not sure if we'll ever need to recursively define groups of producers, but it's implemented and works :)